src/algorithms/peter_norving.py

```diff
@@ -0,0 +1,165 @@
+               for temp_digit in other_values):
+            return values
+        else:
+            return False
+
+    def eliminate(self, values, square, digit):
+        """
+        Eliminate 'digit' from values[square]. Propagate when values or places <= 2.
+
+        Keyword arguments:
+        values -- initial dictionary with values.
+        square -- square positions
+        digit --  selected digit.
+        returns updated solution value list
+        or return False if a contradiction is detected.
+        """
+
+        if digit not in values[square]:
+            return values
+        values[square] = values[square].replace(digit, '')
+        if len(values[square]) == 0:
+            return False
+        elif len(values[square]) == 1:
+            temp_digit = values[square]
+            if not all(self.eliminate(values, temp_square, temp_digit)
+                       for temp_square in self.peers[square]):
+                return False
+        for unit in self.units[square]:
+            #Keys for the sent digit.
+            digit_places = [square for square in unit if digit
+                       in values[square]]
+            if len(digit_places) == 0:
+                return False
+            elif len(digit_places) == 1 and not self.assign(values,
+                    digit_places[0], digit):
+                return False
+        return values
+
+    def search(self, values):
+        """
+        Using depth-first recursive search and propagation,
+        try all possible values in the square before try the next one.
+
+        Keyword arguments:
+        values -- dictionary of possible values {square: digits}, i.e:
+                {'I6': '1379', 'H9': '679', 'I2': '12369' ........
+        returns updated dictionary with final values after search algorithm.
+                {'I6': '1', 'H9': '6', 'I2': '2' ........
+        """
+
+        if values is False:
+            return False
+        if all(len(values[square]) == 1 for square in self.squares):
+            return values
+        (unfilled, square) = min((len(values[square]), square)
+                                 for square in self.squares
+                                 if len(values[square]) > 1)
+
+        return self.get_true_element_from_sequence(
+            self.search(self.assign(values.copy(),
+            square, digit)) for digit in values[square])
+
+    def get_true_element_from_sequence(self, sequence):
+        """
+        returns the element from the generator sequence that has a value ,
+        the dictionary with solved values.
+
+        Keyword arguments:
+        sequence -- values dictionary.
+        """
+
+        for element in sequence:
+            if element:
+                return element
+        return False
+
+    def solve(self, grid):
+        """
+        Solve the sudoku , returns False if the sudoku was invalid.
+
+        Keyword arguments:
+        grid -- initial sudoku grid in string format to solve.
```

Comment:
Where is the documentation for the return value and the example of the grid value received as parameter